File: src/main/java/top/wcpe/wcbot/entity/QQMemberData.java
package top.wcpe.wcbot.entity;

import cn.nukkit.utils.Config;
import cn.nukkit.utils.ConfigSection;
import lombok.Data;

import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;

/**
 * QQ 数据
 *
 * @author: WCPE 1837019522@qq.com
 * @create: 2021-06-22 23:01
 */
@Data
public class QQMemberData {
    private Long qq;
    private String bindPlayerName;
    private boolean isAdmin;
    private Long wcBotPoint;
    private HashMap<String, Long> signDayTimeStamp = new HashMap<>();


    public QQMemberData(Config cfg) {
        this.qq = cfg.getLong("qq");
        this.bindPlayerName = cfg.getString("bindPlayerName");
        this.isAdmin = cfg.getBoolean("isAdmin");
        this.wcBotPoint = cfg.getLong("wcBotPoint");
        ConfigSection signDayTimeStamp = cfg.getSection("signDayTimeStamp");
        if (signDayTimeStamp != null) {
            for (String key : signDayTimeStamp.getKeys(false)) {
                this.signDayTimeStamp.put(key, signDayTimeStamp.getLong(key));
            }
        }
    }

    public Config saveToConfig() {
        Config cfg = new Config();
        cfg.set("qq", this.qq);
        cfg.set("bindPlayerName", this.bindPlayerName);
        cfg.set("isAdmin", this.isAdmin);
        cfg.set("wcBotPoint", this.wcBotPoint);
        for (Map.Entry<String, Long> stringLongEntry : this.signDayTimeStamp.entrySet()) {
            cfg.set("signDayTimeStamp." + stringLongEntry.getKey(), stringLongEntry.getValue());
        }
        return cfg;
    }
}

File: src/main/kotlin/top/wcpe/wcbot/entity/QQMemberData.kt
package top.wcpe.wcbot.entity

import cn.nukkit.utils.Config

/**
 * 由 WCPE 在 2021/11/24 12:43 创建
 * Created by WCPE on 2021/11/24 12:43
 * Github: https://github.com/wcpe
 * QQ: 1837019522
 *
 * @author WCPE
 */
data class QQMemberData(
    val qq: Long,
    var bindGamePlayerName: String,
    var wcBotPoint: Double = 0.0,
    var permissionList: MutableList<String> = mutableListOf(),
    var permissionGroups: MutableList<String> = mutableListOf(),
    var signMonthDataMap: MutableMap<String, SignMonthData> = mutableMapOf()
) {
    constructor(qq: Long, cfg: Config) : this(
        qq,
        cfg.getString("bindGamePlayerName"),
        cfg.getDouble("wcBotPoint"),
        cfg.getStringList("permissionList"), cfg.getStringList("permissionGroups")
    ) {
        if (cfg.exists("signMonthDataMap")) {
            val signMonthDataMapSection = cfg.getSection("signMonthDataMap")
            for (key in signMonthDataMapSection.getKeys(false)) {
                signMonthDataMap[key] = SignMonthData(key, signMonthDataMapSection.getSection(key))
            }
        }
    }

    fun serialize(): Config {
        val cfg = Config()
        cfg["qq"] = qq
        cfg["bindGamePlayerName"] = this.bindGamePlayerName
        cfg["wcBotPoint"] = wcBotPoint
        cfg["permissionList"] = permissionList
        cfg["permissionGroups"] = permissionGroups
        for (entry in signMonthDataMap.entries) {
            cfg["signMonthDataMap.${entry.key}"] = entry.value.serialize()
        }
        return cfg
    }
}